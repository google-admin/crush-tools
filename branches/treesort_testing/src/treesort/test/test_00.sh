test_number=00
description="one column sort"

subtest_desc=("first column"
							"second column using index"
							"first column reversed"
							"second column reversed with index")
subtest_opts=(
'-K AA'
'-k 2 -p'
'-K ^AA'
'-k ^2 -p'
)

for subtest in `seq 0 3`; do
  expected="$test_dir/test_$test_number.$subtest.expected"
  outfile="$test_dir/test_$test_number.$subtest.actual"

  $bin ${subtest_opts[$subtest]} \
       "$test_dir/test.in" \
       > "$outfile"

  if [ $? -ne 0 ] ||
     [ "`diff -q $outfile $expected`" ]; then
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" FAIL
  else
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" PASS
    rm "$outfile"
  fi
done
