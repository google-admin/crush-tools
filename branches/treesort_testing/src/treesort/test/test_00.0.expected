AAþBBþCCþDD
222þ49Uj8LVH5jþ2009-10-02þ13.5003
222þOCS81RVo5jþ2009-10-28þ32.4092
278þtHGNJyQcu1þ2009-10-01þ32.4092
279þafUGgY6q4jþ2009-10-14þ20.7444
279þ0dGw4UOK0Pþ2009-10-26þ17.0268
279þWDEGX5smiWþ2009-10-15þ20.7444
468þvOXGmYLegsþ2009-10-04þ29.5253
468þA-cSNvcqnzþ2009-10-01þ28.4036
577þ4j4n56w__lþ2009-10-25þ14.3872
577þarzhFJlrMBþ2009-10-14þ28.4036
577þ5TcvIxjquUþ2009-10-15þ32.4092
740þOcNUM48IS6þ2009-10-15þ22.8364
740þ7YaqxPKpCRþ2009-10-01þ17.0268
843þZfcHogunHrþ2009-10-01þ28.4036
843þ3XbaqX1keSþ2009-10-02þ32.4092
843þW2OIToxqAMþ2009-10-04þ17.0268
918þ6ff-bhImN7þ2009-10-10þ26.6454
918þz3cqeuch4gþ2009-10-28þ20.7444
965þxVciPqOBiQþ2009-10-14þ17.0268
991þENItEY_hIBþ2009-10-23þ14.9256
